macros/src/main/scala/packaged/insidePackageObject/package.scala

```diff
@@ -2,7 +2,7 @@ package packaged
 
 import scala.gestalt._
 
-package object macros2 {
+package object insidePackageObject {
   inline def plus(a: Int, b: Int): Int = meta {
     q"$a + $b"
   }
```

Comment:
I just published new artifacts of the compiler which fix the problem -- it turns out to be a bug in the compiler. https://github.com/liufengyun/dotty/commit/c232adac117db8678bc1098edf5a17f7776dd78a

Now, the expansion needs to take care of package objects.